tests/test_prolongation.cpp

```diff
@@ -0,0 +1,297 @@
+
+    EXPECT_EQ((int)sol.size(), p_level.m);
+    //testing the embedding. since we embed the coarse gird function, the values of the fine node should be zero
+    for (int j = 0; j < p_level.nr_int + 1; j++) {
+        for (int i = 0; i < p_level.ntheta_int; i++) {
+            if (j % 2 == 0 && i % 2 == 0) {
+                EXPECT_EQ(u_test[(j / 2) * ctheta_int + (i / 2)], sol[j * p_level.ntheta_int + i])
+                    << "The Injection value fails at Index (r,theta): (" + std::to_string(j) + "," + std::to_string(i) +
+                           ")";
+            }
+            else {
+                EXPECT_EQ(sol[j * p_level.ntheta_int + i], 0); //fine nodes set to zero.
+            }
+        }
+    }
+}
+/*!
+ *  \brief Test the extrapolation prolongation operator used in the implicit extrapolation step of the multigrid-cycle. 
+ *  
+ *  The Test creates an arbitrary grid-function on the coarser level and prolongates it. 
+ *  On the fine level we iterate over all nodes and test the result based on whether the node is fine in theta, r or in both.
+ *  The value is determined by a Triangulation of the grid on the coarser level.
+ *  
+ *  Parametrized tests are used to test for different grid sizes and with or without Dirichlet boundary conditions.
+ */
+
+TEST_P(test_prolongation, test_extrapolation_prolongation)
+{
+    gyro::icntl[Param::nr_exp]         = (int)(std::get<0>(GetParam()) / 3) + 3;
+    gyro::icntl[Param::ntheta_exp]     = (std::get<0>(GetParam()) % 3) + 3;
+    gyro::icntl[Param::DirBC_Interior] = std::get<1>(GetParam());
+
+    if (gyro::icntl[Param::nr_exp] == 3)
+        gyro::icntl[Param::fac_ani] = 2;
+
+    gmgpolar test_p;
+
+    create_grid(test_p);
+
+    level& p_level = *(test_p.v_level[0]);
+    int ctheta_int = test_p.v_level[1]->ntheta_int;
+
+    p_level.m  = test_p.v_level[0]->nr * test_p.v_level[0]->ntheta;
+    p_level.mc = test_p.v_level[1]->nr * test_p.v_level[1]->ntheta;
+
+    std::vector<double> u_test(p_level.mc);
+    for (int z = 0; z < p_level.mc; z++) {
+        u_test[z] = z; //arbitrary grid-function to test the prolongation
+    }
+    std::vector<double> sol = p_level.apply_prolongation_ex(u_test);
+
+    for (int j = 0; j < p_level.nr_int + 1; j++) {
+        for (int i = 0; i < p_level.ntheta_int; i++) {
+            if (j % 2 == 0 && i % 2 == 0) { //coarse node injection
+                EXPECT_EQ(u_test[(j / 2) * ctheta_int + (i / 2)], sol[j * p_level.ntheta_int + i])
+                    << "coarse node injection is failing";
+            }
+            else if (i % 2 != 0 && j % 2 == 0) { // theta_i is fine node. upper edge of the triangle
+
+                int i1 = (j / 2) * ctheta_int + (i - 1) / 2;
+                int i2 = (i < p_level.ntheta_int - 1) ? i1 + 1 : (j / 2) * ctheta_int; //cyclic coordinates
+
+                double val = 0.5 * (u_test[i1] + u_test[i2]);
+
+                EXPECT_NEAR(val, sol[j * p_level.ntheta_int + i], 1e-8)
```

Comment:
```suggestion
                EXPECT_NEAR(val, sol[j * p_level.ntheta_int + i], 1e-12)
```
Can we make this succeed with 1e-12 too?